### ISSUE TITLE
Pre-trained model tutorials don't permute image height and width channels

### ISSUE THREAD

**Issue by adrhill** [FluxML/Metalhead.jl#275]
The doc page on [Working with pre-trained models from Metalhead](https://fluxml.ai/Metalhead.jl/dev/tutorials/pretrained/#Training) uses DataAugmentations to crop and normalize an image, however DataAugmentations.jl uses arrays of shape `HWC` instead of `WHCN` like Flux.jl.

Related issue: https://github.com/FluxML/DataAugmentation.jl/issues/56#issuecomment-1934425306

**Comment by adrhill** [comment]
As a quick sanity check, I copied the example from the tutorial and cast the first channel to `Gray`. Since the Images.jl ecosystem uses `HW` matrices of `Colorant`s, the cropped image is oriented "the right way", showing that the code outputs `HWC` data.

```Julia
using Images, ImageInTerminal
using DataAugmentation
using Flux
using Flux: onecold

img = Images.load(download("https://cdn.pixabay.com/photo/2015/05/07/11/02/guitar-756326_960_720.jpg"));

DATA_MEAN = (0.485, 0.456, 0.406)
DATA_STD = (0.229, 0.224, 0.225)

augmentations = CenterCrop((224, 224)) |>
                ImageToTensor() |>
                Normalize(DATA_MEAN, DATA_STD);

data = apply(augmentations, Image(img)) |> itemdata;
Gray.(data[:, :, 1])
```

<img width="965" alt="image" src="https://github.com/FluxML/Metalhead.jl/assets/20258504/1dfa1320-d1c4-4f37-a218-5a338f264805">

**Comment by ToucheSir** [comment]
This is a tricky one because interop requirements pull us in conflicting directions. You've identified how Julia image libraries expect `HWC`. We use `WHCN` because underlying libraries like cuDNN use `NCHW`. Technically, I'm not aware of any routines in said libraries which would behave differently if you swapped the height and weight dims, so that can be worked around.

What can't be worked around is interop on pretrained weights. Those come from Python libraries, and Python libraries are using `NCHW` or `NWHC`. We _might_ be able to do some transformations to make them work for `NCWH`/`HWNC`, but without spending a bunch of time to look into that I don't want to say for sure.

**Comment by adrhill** [comment]
> What can't be worked around is interop on pretrained weights. Those come from Python libraries, and Python libraries are using `NCHW` or `NWHC`. We might be able to do some transformations to make them work for `NCWH`/`HWNC`, but without spending a bunch of time to look into that I don't want to say for sure.

This confused me a bit. I thought all Metalhead models expect `WHCN`. Is this not the case?

**Comment by adrhill** [comment]
Ok, then maybe I didn't clarify my issue well enough:

`data`, the output of the DataAugmentation pipeline in the documentation, is in `HWC` format and fed into a model that expects `WHCN` inputs. I would consider this a "bug" in the documentation. Users copying this example would use inputs with permuted dimensions and most likely get worse predictions.

This could be addressed by adding a call to `permutedims` in the example.
I've also opened a PR for a matching transformation in DataAugmentations.jl: https://github.com/FluxML/DataAugmentation.jl/pull/89

**Comment by adrhill** [comment]
Alternatively, an argument could be made that DataAugmentation.jl's `ImageToTensor` should return `WHC` arrays instead of the current `HWC`.

```Julia
using DataAugmentation, Images

h, w = 20, 30
im = rand(RGB, h, w)

item = Image(im)
tfm = ImageToTensor()
apply(tfm, item) |> itemdata |> size
```

<img width="544" alt="image" src="https://github.com/FluxML/Metalhead.jl/assets/20258504/d49c9cd3-d4c5-4b15-bd0c-72c6dbab0d2a">
